src/lib/operators/abstract_operator.cpp

```diff
@@ -33,7 +35,14 @@ void AbstractOperator::execute() {
     _output = _on_execute(transaction_context);
     transaction_context->on_operator_finished();
   } else {
-    _output = _on_execute(transaction_context);
+    if (dynamic_cast<SQLQueryOperator*>(this)) {
+      // TODO(anyone): Once the SQL Query Operator is refactored (#304), this should be removed
+      _output = _on_execute(nullptr);
```

Comment:
I’d just use this call for every operator. Then the friend declaration could be removed.

Reply:
done